x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunker.java

```diff
@@ -68,7 +68,7 @@ public static EmbeddingType fromDenseVectorElementType(DenseVectorFieldMapper.El
     private final EmbeddingType embeddingType;
     private final ChunkingSettings chunkingSettings;
 
-    private List<List<String>> chunkedInputs;
+    private List<ChunkOffsetsAndInput> chunkedOffsets;
```

Comment:
`ChunkOffsetsAndInput` will eventually become just `ChunkOffsets` once we don't need to include the chunked text in the inference results

Reply:
Makes sense, just for my own knowledge, do we have a plan for how we'd remove the text from the inference results? Seems like it might be a breaking change unless we write some logic to still display the text even only the offsets are stored.

Reply:
The next step will be to update the inference results to return both the offsets and the strings. Then semantic text will adapt to that change and the strings can be removed. This function isn't exposed to a REST API it is only used internally so it is possible to safely make the change without breaking